src/gobupload/storage/update_table.py

```diff
@@ -431,232 +418,68 @@ def _get_query(self):
 
-        :param event:
-        :param include_gob_id:
-        :return:
-        """
-        result = []
-        for db_field, row_key in self.fields.items():
-            if db_field == FIELD.GOBID:
-                if include_gob_id:
-                    # Add without quotation marks, _gobid is the only int in the row
-                    result.append(str(event.get(row_key)) if event.get(row_key) else 'NULL')
-                continue
-
-            if row_key is None:
-                result.append('NULL')
-            else:
-                result.append(f"'{event.get(row_key)}'{self._cast_expr(db_field)}" if event.get(row_key) else 'NULL')
-        return result
-
-    def _column_list(self, include_gob_id=False):
-        """Returns the list of columns for an insert/update query
-
-        :param include_gob_id:
-        :return:
-        """
-        return ','.join([key for key in self.fields.keys() if not (key == FIELD.GOBID and include_gob_id is False)])
-
-    def _write_events(self, queue):
-        """Writes events in queue to the database. Events in queue should all be of the same type (add/modify/delete).
-
-        :param queue:
-        :return:
-        """
-        if not queue:
-            return 0
-
-        if queue[0]['event_type'] == ADD.name:
-            self._write_add_events(queue)
-        elif queue[0]['event_type'] == MODIFY.name:
-            self._write_modify_events(queue)
-        elif queue[0]['event_type'] == DELETE.name:
-            self._write_delete_events(queue)
-
-        self.update_cnt += len(queue)
-
-        queue.clear()
-
-    def _write_add_events(self, events):
-        """Updates database with adds
-
-        :param events:
-        :return:
-        """
-        values = ",\n".join([f"({','.join(self._values_list(dict(event)))})" for event in events])
-        query = f"INSERT INTO {self.relation_table} ({self._column_list()}) VALUES {values}"
-        return _execute(query)
-
-    def _write_modify_events(self, events):
-        """Updates database with modifies
-
-        :param events:
-        :return:
-        """
-        values = ",\n".join([f"({','.join(self._values_list(dict(event), True))})" for event in events])
-
-        set_values = ',\n'.join([f"{col_name} = v.{col_name}{self._cast_expr(col_name)}"
-                                 for col_name in self.fields.keys() if col_name != FIELD.GOBID])
-
-        query = \
-            f"UPDATE {self.relation_table} AS rel\n" \
-            f"SET {set_values}\n" \
-            f"FROM (VALUES {values}) AS v({self._column_list(True)})\n" \
-            f"WHERE rel.{FIELD.GOBID} = v.{FIELD.GOBID}"
-
-        return _execute(query)
-
-    def _write_delete_events(self, events):
-        """Updates database with deletes
-
-        :param events:
-        :return:
-        """
-        ids = ','.join([str(event['rel__gobid']) for event in events])
+        return relation
```

Comment:
return {
    field: datetime.combine(relation.get(field), datetime.min.time())
    for field in timestamps
    if relation.get(field) and isinstance(relation.get(field), date)
}

Reply:
I knew it ;) Updated (spot the difference)